src/fsharp/FSharp.Compiler.Service.ProjectCracker/ProjectCracker.fs

```diff
@@ -20,7 +20,7 @@ type ProjectCracker =
             let referencedProjects = Array.map (fun (a, b) -> a, convert b) opts.ReferencedProjectOptions
             
             let sourceFiles, otherOptions = 
-                opts.Options |> Array.partition (fun x -> Path.GetExtension(x).ToLower() = ".fs")
+                opts.Options |> Array.partition (fun x -> x.IndexOfAny(Path.GetInvalidPathChars()) = -1 && Path.GetExtension(x).ToLower() = ".fs")
```

Comment:
Looks like you can use ``not (FileSystem.IsInvalidPathShim(x))`` like this https://github.com/Microsoft/visualfsharp/blob/379f729dca9f310b474e3edf0669c71fe39b0129/src/fsharp/vs/IncrementalBuild.fs#L1612